samples/NativeIISSample/Startup.cs

```diff
@@ -60,9 +61,35 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, IAuthent
                     await context.Response.WriteAsync(key + ": " + value + Environment.NewLine);
                 }
                 await context.Response.WriteAsync(Environment.NewLine);
+
+                var serverVariables = context.Features.Get<IServerVariablesFeature>();
+
+                if (serverVariables != null)
+                {
+                    // accessing IIS server variables
+                    await context.Response.WriteAsync("Server Variables:" + Environment.NewLine);
+
+                    for (var i = 0; i < IISServerVarNames.Length; i++)
+                    {
+                        var varName = IISServerVarNames[i];
+                        await context.Response.WriteAsync(varName + ": " + serverVariables[varName] + Environment.NewLine);
+                    }
+                }
             });
         }
 
+        private static readonly string[] IISServerVarNames =
+        {
+            "AUTH_TYPE",
```

Comment:
We should consider putting IIS Server Variable Names in HttpAbstractions. UrlRewrite uses them too: https://github.com/aspnet/BasicMiddleware/blob/dev/src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/ServerVariables.cs

Reply:
Uhhh nah. These are IIS specific.  If you want to share code, crate a sources package but I don’t think it’s worth it

Reply:
And this is only a sample.